marshmallow_recipe/serialization.py

```diff
@@ -36,7 +36,9 @@ def __call__(
 
 
 class DumpFunction(Protocol):
-    def __call__(self, data: Any, *, naming_case: NamingCase | None = None) -> dict[str, Any]: ...
+    def __call__(
+        self, data: Any, *, naming_case: NamingCase | None = None, t: type | None = None
```

Comment:
What is the point of passing t here?

Reply:
that function https://github.com/anna-money/marshmallow-recipe/pull/163/commits/c396fa80d7cff0c988272a6784eb77f9a16b80b2#diff-7f5a7271c4eef44b2f5fda6334ecae0952550de169520dde4a919d86949361c2R655

and that test https://github.com/anna-money/marshmallow-recipe/pull/163/commits/c396fa80d7cff0c988272a6784eb77f9a16b80b2#diff-7f5a7271c4eef44b2f5fda6334ecae0952550de169520dde4a919d86949361c2R655

should give an answer

the issue is that `type(instance)` returns unsubscripted (open) generic but we can get subscripted (closed) generic from `instance.__origin_class__` but it does not for `frozen=True` or `slots=True` dataclasses that is why we need to specify type explicitly

Reply:
PR description updated